crates/re_viewer/src/ui/blueprint_sync.rs

```diff
@@ -0,0 +1,202 @@
+use arrow2_convert::field::ArrowField;
+use re_data_store::LogDb;
+use re_log_types::{
+    Component, DataCell, DataRow, EntityPath, RowId, SerializableComponent, TimePoint,
+};
+use re_viewer_context::SpaceViewId;
+
+use crate::blueprint_components::{
+    panel::PanelState,
+    space_view::SpaceViewComponent,
+    viewport::{
+        AutoSpaceViews, SpaceViewMaximized, SpaceViewVisibility, ViewportLayout, VIEWPORT_PATH,
+    },
+};
+
+use super::{Blueprint, SpaceView, Viewport};
+
+// TODO(jleibs): Put this somewhere more global
+pub fn store_one_component<C: SerializableComponent>(
```

Comment:
Seems like `re_query` is a good place, despite its name, since it is a mirror image of `query_single_component`

Reply:
I actually put both in re_data_store util since they don't have a dep on re_query.